packages/buidler-core/src/internal/buidler-evm/jsonrpc/server.ts

```diff
@@ -0,0 +1,46 @@
+import debug from "debug";
+import * as http from "http";
+
+import { EthereumProvider } from "../../../types";
+
+import JsonRpcHandler from "./handler";
+
+const log = debug("buidler:core:buidler-evm:jsonrpc");
+
+export interface JsonRpcServerConfig {
+  hostname: string;
+  port: number;
+
+  ethereum: EthereumProvider;
+}
+
+export class JsonRpcServer {
+  private _config: JsonRpcServerConfig;
+
+  constructor(config: JsonRpcServerConfig) {
+    this._config = config;
+  }
+
+  public listen = (): Promise<number> => {
+    return new Promise<number>((resolve, reject) => {
+      const { hostname, port, ethereum } = this._config;
+
+      log(`Starting JSON-RPC server on port ${port}`);
+
+      const handler = new JsonRpcHandler(ethereum);
+      const server = http.createServer(handler.requestListener);
+
+      process.once("SIGINT", async () => {
+        log(`Stopping JSON-RPC server`);
+
+        resolve(0);
+      });
+
+      process.once("uncaughtException", reject);
```

Comment:
I don't really get why this is here. It seems a little too general/risky.

Reply:
Removing. We should have error handling in all the right places anyways.